pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -321,17 +375,38 @@ func (p *provider) Create(machine *clusterv1alpha1.Machine, _ *cloudprovidertype
 	// because its pod got deleted
 	// The secret has an ownerRef on the VMI so garbace collection will take care of cleaning up
 	terminationGracePeriodSeconds := int64(30)
-	userDataSecretName := fmt.Sprintf("userdata-%s-%s", machine.Name, strconv.Itoa(int(time.Now().Unix())))
-	requestsAndLimits, err := parseResources(c.CPUs, c.Memory)
+	userDataSecretName := fmt.Sprintf("userdata-%s-%s", c.VirtualMachineName, strconv.Itoa(int(time.Now().Unix())))
+
+	resourceRequirements := kubevirtv1.ResourceRequirements{}
+	labels := map[string]string{"kubevirt.io/vm": c.VirtualMachineName}
```

Comment:
Before we had `machine.Name` , which was unique for each `machine` (if replicas=3 in MD, we have 3 machines, each having a name).
Here `c.VirtualMachineName` is common to all machines, as comming from the MD.
Is it really what we want ?

It's a good idea to have a common label among all machines from the same MD (we would also need it for the kubermatic/kubermatic#8742), but the label key should contain `md` and not `vm`.
It's also a good idea to keep `kubevirt.io/vm` = `machine.Name`
having both makes sense.